server/src/main/java/org/elasticsearch/health/metadata/HealthDiskThresholdSettings.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.health.metadata;
+
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.RelativeByteSizeValue;
+import org.elasticsearch.common.util.DiskThresholdSettingParser;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+/**
+ * A container to keep the settings for health disk thresholds up to date with cluster setting changes.
+ * These thresholds are used to determine if a node that does not hold data has enough disk space.
+ */
+public class HealthDiskThresholdSettings {
+
+    public static final Setting<String> CLUSTER_HEALTH_DISK_YELLOW_THRESHOLD_SETTING = new Setting<>(
+        "cluster.health.disk.threshold.yellow",
```

Comment:
nit: Should we keep the low/high terminology our users are used to already? ie. "cluster.health.reporting.disk.threshold.low"

Reply:
I think we should also document these settings

Reply:
+1. I just didn't want to document until they are not behind a feature flag. I was planning to add them in a separate PR.